### DISCUSSION TITLE
ganache-cli address same as instructor? ///* Lesson 5 ///* 4:22:00

### DISCUSSION THREAD

**Discussion by oliveruf42** [smartcontractkit/full-blockchain-solidity-course-py#984] [Q&A] [answered]
Hey guys i have a question about my test addresses, and private keys in the ```ganache-cli``` I entered 
```ganache-cli --deterministic``` and from my understanding this command would pull up the same private keys and addresses each time, instead of randomly generated ones. When i started to copy it into VSCode i noticed that i have the exact same test keys and addresses as Patrick in the video. I've tried multiple searches on Stack Overflow , here, and Google. However I can't seem to find anyone with the same problem. I just want to make sure that I'm not about to alter the course instructors test contracts. Thanks in advance.

```
web3_py_simple_storage CuervoJones$ ganache-cli --deterministic
Ganache CLI v6.12.2 (ganache-core: 2.13.2)

Available Accounts
==================
(0) 0x90F8bf6A479f320ead074411a4B0e7944Ea8c9C1 (100 ETH)
(1) 0xFFcf8FDEE72ac11b5c542428B35EEF5769C409f0 (100 ETH)
(2) 0x22d491Bde2303f2f43325b2108D26f1eAbA1e32b (100 ETH)
(3) 0xE11BA2b4D45Eaed5996Cd0823791E0C93114882d (100 ETH)
(4) 0xd03ea8624C8C5987235048901fB614fDcA89b117 (100 ETH)
(5) 0x95cED938F7991cd0dFcb48F0a06a40FA1aF46EBC (100 ETH)
(6) 0x3E5e9111Ae8eB78Fe1CC3bb8915d5D461F3Ef9A9 (100 ETH)
(7) 0x28a8746e75304c0780E011BEd21C72cD78cd535E (100 ETH)
(8) 0xACa94ef8bD5ffEE41947b4585a84BdA5a3d3DA6E (100 ETH)
(9) 0x1dF62f291b2E969fB0849d99D9Ce41e2F137006e (100 ETH)


```

**Comment by cromewar** [comment, marked as answer]
Hello @oliveruf42 
Don't worry about it, ganache creates the local blockchain which exists just on your computer and nowhere else, and setting up as deterministic gives everyone the exact same accounts and private keys, this is on purpose. 
Let's suppose I'm working on a project and I want you to test it, in order to you for get the exact same results as I'm getting on my computer is necessary the variables remain the same, so if we use `--deterministic` we will reproduce the exact same results.

Summarizing, this is not a bug... is a feature.

Hope this info might be helpful, best regards. Cromewar.

**Comment by oliveruf42** [comment]
Ok cool ,thanks for the response !
